nomad/deploymentwatcher/deployments_watcher_test.go

```diff
@@ -569,14 +579,36 @@ func TestWatcher_AutoPromoteDeployment(t *testing.T) {
 		return a
 	}
 
-	a := alloc()
-	b := alloc()
+	rollingAlloc := func() *structs.Allocation {
+		a := mock.Alloc()
+		a.DeploymentID = d.ID
+		a.CreateTime = now.UnixNano()
+		a.ModifyTime = now.UnixNano()
+		a.TaskGroup = "api"
+		a.AllocatedResources.Tasks["api"] = a.AllocatedResources.Tasks["web"].Copy()
+		delete(a.AllocatedResources.Tasks, "web")
+		a.TaskResources["api"] = a.TaskResources["web"].Copy()
+		delete(a.TaskResources, "web")
+		a.DeploymentStatus = &structs.AllocDeploymentStatus{
+			Canary: false,
+		}
+		return a
+	}
+
+	// Web taskgroup (0)
+	a := canaryAlloc()
+	b := canaryAlloc()
+
+	// Api taskgroup (1)
+	c := rollingAlloc()
+	e := rollingAlloc()
```

Comment:
Might want to give up on 1 letter variable names at this point (`ca1`, `ca2`, `ra1`, and `ra2` perhaps), but not a blocker.

Reply:
Thank you for taking a look at this @schmichael and updating the changelog! Nice, that was feeling a bit weird and I'll make that change shortly